### DISCUSSION TITLE
[Weather refresh] Add check if location services are enabled

### DISCUSSION THREAD

**Discussion by min7-i** [breezy-weather/breezy-weather#1447] [Ideas]
### Describe your idea

This is a follow-up to #1413.

It's possible to check if location services are enabled independently from location permissions. Hence, the idea is to
- add a separate refresh error type (snackbar) for it with an action to open location system settings
- add a check if location services are enabled in refresh helper

**Approach**
```
val locationManager = Context.getSystemService(Context.LOCATION_SERVICE) as LocationManager
LocationManagerCompat.isLocationEnabled(locationManager)
```

### Other details

It seems that `PermissionChecker.checkSelfPermission` which is used in [Context.hasPermission](https://github.com/breezy-weather/breezy-weather/blob/46fb592b0693bc2ba1444aa0e6cefad9f159075c/app/src/main/java/org/breezyweather/common/extensions/ContextExtensions.kt#L46-L48) only returns the proper results for the location permissions when location is enabled. However, when using `ContextCompat.checkSelfPermission` instead, it always returns the correct result. That way, we can properly separate the checks for location permissions from checking if location services are enabled.

I can submit a PR for this.

### Acknowledgements

- [X] I have searched the [existing issues](https://github.com/breezy-weather/breezy-weather/issues) and [existing discussions](https://github.com/breezy-weather/breezy-weather/discussions/categories/ideas) and this is a new ticket, **NOT** a duplicate or related to another open OR closed issue or discussion.
- [X] The idea I’m suggesting is not in the [list of features that will not be implemented](https://github.com/breezy-weather/breezy-weather/blob/main/README.md#features-that-will-not-be-implemented).
- [X] I have updated the app to **[version 5.2.8 or later](https://github.com/breezy-weather/breezy-weather/releases/latest)**. Even if I’m getting updates from F-Droid default repo, I don’t need to uninstall/reinstall to update.
- [X] I wrote my request in English, and any screenshot I attached are also in English (language can be changed from Settings > Appearance). I can write in my native language under the English text, if necessary.
- [X] I understand that this app is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY, so I may not get a positive reply to my request, if any at all.

**Comment by papjul** [comment]
Thank you for your investigation of the issue, I will happily review it x)

**Comment by min7-i** [comment]
Implemented in #1467. _Apparently, mentioning discussions in PRs doesn't automatically close them._
